src/precice/tests/ParallelTests.cpp

```diff
@@ -742,6 +742,291 @@ BOOST_AUTO_TEST_CASE(TestBoundingBoxInitialization)
   precice.finalize();
 }
 
+// Test case for parallel mesh partitioning without any mapping. Each solver
+// runs on two ranks. SolverTwo defines 5(2 and 3) vertices which need to be
+// repartitioned on SolverOne according to the defined boundingBoxes
+// (resulting in 3 and 2 vertices per rank. The boundingBoxes don't have any
+// overlap.
+BOOST_AUTO_TEST_CASE(BoundingBoxExplicitNoOverlap)
+{
+  PRECICE_TEST("SolverOne"_on(2_ranks), "SolverTwo"_on(2_ranks));
+  std::string configFilename = _pathToTests + "explicit-bounding-box.xml";
+
+  if (context.isNamed("SolverOne")) {
+    // Defines the bounding box and writes data to the received mesh
+    SolverInterface interface(context.name, configFilename, context.rank, context.size);
+    const int       otherMeshID = interface.getMeshID("MeshTwo");
+    const int       dataID      = interface.getDataID("Velocities", otherMeshID);
+    const int       dim         = interface.getDimensions();
+
+    std::vector<double> boundingBox = context.isMaster() ? std::vector<double>({0.0, 1.0, 0.0, 3.5}) : std::vector<double>{0.0, 1.0, 4.0, 7};
+    // Set bounding box
+    interface.setBoundingBoxes(otherMeshID, boundingBox.data(), 1);
+    // Initialize the solverinterface
+    double dt = interface.initialize();
+
+    // Get relevant size, allocate data structures and retrieve coordinates
+    const int meshSize = interface.getMeshVertexSize(dataID);
+
+    // According to the bounding boxes and vertices: the master rank receives 3 vertices, the slave rank 2
+    const bool expectedSize = (context.isMaster() && meshSize == 3) || (!context.isMaster() && meshSize == 2);
+    BOOST_TEST(expectedSize);
+
+    // Allocate memory
+    std::vector<int>    ids(meshSize);
+    std::vector<double> coordinates(meshSize * dim);
+    interface.getMeshVerticesWithIDs(otherMeshID, meshSize, ids.data(), coordinates.data());
+
+    // Check the received vertex coordinates
+    std::vector<double> expectedPositions = context.isMaster() ? std::vector<double>({0.0, 1.0, 0.0, 2.0, 0.0, 3.0}) : std::vector<double>({0.0, 4.0, 0.0, 5.0});
+    BOOST_TEST(expectedPositions == coordinates);
+
+    // Check the received vertex IDs (IDs are local?!)
+    std::vector<int> expectedIDs;
+    for (int i = 0; i < meshSize; ++i)
+      expectedIDs.emplace_back(i);
+    BOOST_TEST(expectedIDs == ids);
+
+    // Create some unique writeData in order to assert it in the other participant
```

Comment:
```suggestion
    // Create some unique writeData in order to check it in the other participant
```
I am picky, but just to show again that we use the term "assert" differently.

Reply:
I fixed it although I don't really get how this is related to the API and the applied macros.